lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2383,6 +2383,17 @@ protected static function compute_style_properties( $styles, $settings = array()
 
 			// Processes background styles.
 			if ( 'background' === $value_path[0] && isset( $styles['background'] ) ) {
+				/*
+				 * For user-uploaded images at the block level, assign defaults.
+				 * Matches defaults applied in the editor and in block supports: background.php.
+				 */
+				if ( static::ROOT_BLOCK_SELECTOR !== $selector && ! empty( $styles['background']['backgroundImage']['id'] ) ) {
```

Comment:
This is the same as the assignments performed in [background.php](https://github.com/WordPress/gutenberg/blob/8aad1dc460825710fa6a9e136d0ffb393648b5b8/lib/block-supports/background.php#L61-L67)

Thinking this could be abstracted to a function either in background.php, or the style engine? 

What do you reckon @andrewserong ?

Reply:
> or the style engine?

Well, I think the style engine shouldn't provide defaults willy-nilly, but maybe it could "register" defaults or accepts them in the options argument. 

Maybe a separate function is better here, e.g., 

```php

gutenberg_style_engine_get_styles( array( 'background' => gutenberg_apply_style_defaults( $styles['background'] ) ) )

```

Reply:
To be honest, I think I'd go with the duplication for now or until we have another block support that needs default values in this way, as what we're doing with background image seems fairly unique right now. Are there other use cases for `gutenberg_apply_style_defaults` we know of in the near-term?

If not, I'd lean a little toward going with the duplication, especially since it's only ~5 lines or so, and revisit if and when we have other defaults to apply. What do you think?

Reply:
> Are there other use cases for gutenberg_apply_style_defaults we know of in the near-term?

Not that I know. 

Thanks for the feedback. I agree with your reasoning. I'll leave it for now then. 🙇🏻